sky/task.py

```diff
@@ -295,10 +296,24 @@ def from_yaml(yaml_path: str) -> 'Task':
 
         task_storage_mounts: Dict[str, storage_lib.Storage] = {}
         all_storages = fm_storages
+        enabled_clouds = global_user_state.get_enabled_clouds()
+        enabled_clouds = [str(cloud) for cloud in enabled_clouds]
+        r2_is_enabled, _ = cloudflare.check_credentials()
+        if r2_is_enabled:
+            enabled_clouds.append('r2')
         for storage in all_storages:
             mount_path = storage[0]
             assert mount_path, \
                 'Storage mount path cannot be empty.'
+            store_type = storage[1]['store']
+            if store_type:
+                cloud_type = storage_lib.STORE_TYPE_TO_CLOUD_TYPE[store_type]
+                if cloud_type not in enabled_clouds:
+                    with ux_utils.print_exception_no_traceback():
+                        raise exceptions.CloudDisabledError(
+                            f'Storage \'store:{store_type}\' specified, but '
+                            f'\'{cloud_type}\' access is disabled. Enable '
+                            f'\'{cloud_type}\' to fix.')
```

Comment:
I think this check should happen in storage.py, possibly in the `Store._validate()` method. Reasons include a) storage object may be initialized outside of task context b) YAML is not the only way to initialize storage/task objects, user can use python API too.

Reply:
Sounds great. I can create a function called `get_enabled_storage_cloud()` and call it in `Store._validate()` to replace this portion. And it seems like the way how it would be implemented can be determined after deciding the matter discussed in this issue #1943

Reply:
Sounds good. While #1943 would require a longer term fix, let's go ahead and add `get_enabled_storage_clouds()` to `global_user_state.py`, which effectively does:
```
def get_enabled_storage_clouds():
   clouds = get_enabled_clouds()
   clouds.filter(STORE_ENABLED_CLOUDS) # Exclude lambda, IBM etc.
   if is_r2_enabled:
      clouds.append(r2)
   return clouds
```

We can then call this method to get the list of enabled storage clouds.

Reply:
Resolved